ext/serializer.c

```diff
@@ -1256,6 +1256,25 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         }
 
         add_assoc_zval(el, "service", &prop_service_as_string);
+
+        // Add _dd.base_service if service name differs from mapped root service name
+        zval *prop_root_service = &span->root->property_service;
+        ZVAL_DEREF(prop_root_service);
+        zval prop_root_service_as_string;
+        ddtrace_convert_to_string(&prop_root_service_as_string, prop_root_service);
```

Comment:
```suggestion
        zval prop_root_service_as_string;
        ddtrace_convert_to_string(&prop_root_service_as_string, &span->root->property_service);
```
ddtrace_convert_to_string is able to handle refs directly